manager/allocator/networkallocator/networkallocator.go

```diff
@@ -764,7 +778,15 @@ func (na *NetworkAllocator) allocatePools(n *api.Network) (map[string]string, er
 		}
 		pools[poolIP.String()] = poolID
 
-		gwIP, _, err := ipam.RequestAddress(poolID, net.ParseIP(ic.Gateway), nil)
+		if dOptions == nil {
+			dOptions = make(map[string]string)
+		}
+		dOptions[ipamapi.RequestAddressType] = netlabel.Gateway
+		// set ipam allocation method to serial
+		dOptions = setIPAMSerialAlloc(dOptions)
+		defer delete(dOptions, ipamapi.RequestAddressType)
+
+		gwIP, _, err := ipam.RequestAddress(poolID, net.ParseIP(ic.Gateway), dOptions)
```

Comment:
@fcrisciani in the master branch, this statement seems to be inside a conditional:
```
if ic.Gateway != "" || gwIP == nil {
...
}
```
Do we need that here?

Reply:
good point, have to check

Reply:
Looking at it better looks like that logic with the if seems had been introduced by another commit https://github.com/docker/swarmkit/commit/fb74191488a7b37c364f528799c390e4c3a35d7e, that is a major restructure looks like and the logic there extract gwIP.

This change as is does not change the current behavior but simply pass the additional flag to the IPAM.

Reply:
Sounds good, thanks for checking!